### DISCUSSION TITLE
Non interactive mode?

### DISCUSSION THREAD

**Discussion by Lucas-C** [ClementTsang/bottom#1097] [Q&A] [answered]
Hi!

I have just discovered (coming from `gotop` and then `ytop`) & tested `bottom`, and it is great!
Thank you for maintaining it 😊

I am looking for a non interactive tool that provides the list of Windows processes currently running, with their CPU & memory usage
(and ideally limit this list to 10 or 20 rows).

I haven't found any related option in the documentation ( https://clementtsang.github.io/bottom/nightly/configuration/command-line-flags/ ), but I dare to ask: does `bottom` has a non interactive (aka "single-print-on-terminal") mode?
And if no, would you recommend any altternative by chance?

**Comment by ClementTsang** [comment]
Sorry, but I'm not totally sure what you're asking. Could using the config to only display processes be fine? As for only specific columns, that's an upcoming feature.

**Reply by Lucas-C** [reply]
> Sorry, but I'm not totally sure what you're asking. Could using the config to only display processes be fine? As for only specific columns, that's an upcoming feature.

I have tried that already, using this TOML configuration:
```toml
[[row]]
  [[row.child]]
    type="proc"
    default=true
```

It's great, but I'd like to run a `bottom` in the context of a Windows CI pipeline, to have grasp of what processes are currently eating up CPU / RAM. That is why I need a non-interactive mode.

Is that clearer?

**Reply by ClementTsang** [reply, marked as answer]
Hmm. So you just want it to return a single snapshot of process usage, is that correct?

As I might actually have something that might do that planned, though it was more in the context as a debug tool for bug reports.

**Reply by Lucas-C** [reply]
Sounds interesting!

Thank you for your answer 😊
